### PR TITLE
bs4 plan shopping family info & choose cov for HH pages

### PR DESCRIPTION
**Pull request by scaustin34**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: 
https://www.pivotaltracker.com/story/show/187141782
https://www.pivotaltracker.com/story/show/187141800

# A brief description of the changes

Current behavior:

New behavior:

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/views/shared/_plan_shopping_progress.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<% step = (defined? step) ? step.to_i : 1 %>
+<% ridp = (defined? ridp) ? ridp : false %>
```
Comment (charlienparker) [inline app/views/shared/_plan_shopping_progress.html.erb:2]: are either `continue_disabled` or `ridp` being used here? If not, I think we can remove them...

and if we do remove them, I'm thinking this partial ends up looking exactly like the `consumer_signup` partial we have, save for the "title" label. would it be worth combining these two partials into one "progress_nav" partial or something to that effect?

File: app/views/insured/family_members/index.html.erb [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
-      </div>
+    </div>
+  </div>
+
+  <div id="plan-shopping-page-navigation-buttons" class="d-flex flex-column flex-md-row mt-4" >
+    <% group_selection_url = new_insured_group_selection_path(person_id: @person.id, employee_role_id: @employee_role.try(:id), consumer_role_id: @consumer_role.try(:id), change_plan: @change_plan, market_kind: @market_kind, sep_id: @sep.try(:id), qle_id: @qle.try(:id), effective_on_option_selected: params[:effective_on_option], resident_role_id: @resident_role.try(:id)) %>
+    <% find_sep_url = find_sep_insured_families_path(employee_role_id: @employee_role.try(:id), consumer_role_id: @consumer_role.try(:id), change_plan: @change_plan)%>
+    <% url = (@consumer_role.present? && !is_under_open_enrollment? ? find_sep_url : group_selection_url) %>
+
+    <%= render partial: "shared/progress_navigation_buttons", locals: {previous_style: "outline", next_link: url } %>
```
Comment (charlienparker) [inline app/views/insured/family_members/index.html.erb:16]: we can probably remove the previous_style argument here - it defaults to "outline" in the partial

File: app/helpers/navigation_helper.rb [hunk 3]
```diff
@@ -34,11 +34,99 @@ def local_assigned_boolean(local, default)
...
+    if @change_plan.present?
+      {step: '1', option_hash: qle_nav_options}
+    elsif @type == "employee"
+      {step: '4', option_hash: sep_nav_options}
+    else
+      {step: '3', option_hash: individual_nav_options}
+    end
+  end
+
+  def family_info_progress_hash
```
Comment (charlienparker) [inline app/helpers/navigation_helper.rb:47]: this def is repeated from the above

**Review by charlienparker — CHANGES_REQUESTED** [review]

File: app/helpers/navigation_helper.rb (hunk 3 above)
Comment (scaustin34) [reply app/helpers/navigation_helper.rb]: all good catches, thanks!

File: app/helpers/navigation_helper.rb [hunk 4]
```diff
@@ -34,10 +34,88 @@ def local_assigned_boolean(local, default)
...
+    elsif @change_plan == "change_plan"
+      if (@market_kind == "individual" && !is_under_open_enrollment?) || @enrollment_kind == 'sep'
+        {step: '2', option_hash: shop_for_plans_nav_options}
+      else
+        {step: '1', option_hash: sep_shop_for_plans_nav_options}
+      end
+    end
+  end
+
   def sign_up_nav_options
```
Comment (charlienparker) [inline app/helpers/navigation_helper.rb:65]: nitpick - can we move this def above the progress_hash helpers? maybe even add comments marking which of these defs belong to the signup flow and which of these defs belong to the coverage experiences? just a little jarring to me to have sign_up_nav_options be between a bunch of defs for a separate experience...

**Review by charlienparker — APPROVED** [review]

File: app/controllers/insured/family_members_controller.rb [hunk 5]
```diff
@@ -130,6 +133,8 @@ def show
     authorize @family, :show?
 
     set_view_person
+
+    @bs4 = true if params[:bs4] === "true"
```
Comment (charlienparker) [inline app/controllers/insured/family_members_controller.rb:137]: is triple equals intended here?

**Review by charlienparker — APPROVED** [review]

File: db/seedfiles/translations/en/me/shared.rb [hunk 6]
```diff
@@ -56,7 +56,9 @@
...
...coverage: <br> <ul> <li>Yourself</li><li>Your spouse if you’re legally married (even if you don’t live together)</li><li>Everyone claimed as a dependent on the tax return</li><li>Any child under 21 who lives with you, even if you don’t claim them as a dependent</li><li>Optionally, list any of your children up to age 26, even if they don’t live with you, if you’d like to include them in your health coverage application</li></ul> If you file a federal tax return, include the following people. Include everyone on this list even if some people don’t need health coverage: <br> <ul> <li>Yourself</li> <li> Your spouse if you’re legally married (even if you don’t live together) </li> <li> Anyone under the age of 21 who lives with you</li> <li> Optionally, list any of your children up to age 26, even if they don’t live with you, if you’d like to include them in your health coverage application </li> </ul> You don’t need to list: <ul> <li> Roommates</li> <li> Your unmarried domestic partner (unless you claim this person as your tax dependent)</li> <li> Your spouse if you’re legally separated, divorced, or are a victim of domestic abuse, domestic violence, or spousal abandonment</li> </ul> ",
```
Comment (kristinmerbach) [inline db/seedfiles/translations/en/me/shared.rb:61]: We should not be adding to the markup in translation issue, please separate out.

File: app/views/ui-components/v1/modals/_detail_modal.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<div class="modal fade" id="plan-details-modal">
+  <div class="modal-dialog modal-lg">
+    <div class="modal-content">
+      <div class="modal-header" style="border-bottom: none">
```
Comment (kristinmerbach) [inline app/views/ui-components/v1/modals/_detail_modal.html.erb:4]: no inline styles please, use `border-bottom-0` class

File: app/views/ui-components/bs4/v1/navs/_header.html.erb [hunk 8]
```diff
@@ -1,8 +1,8 @@
 <hidden id='dchbx_enroll_date_of_record' class='d-none'><%= TimeKeeper.date_of_record.iso8601 %></hidden>
 <header class="bg-white navbar-primary py-2 <%= 'pre_prod_nav_color' if ENV['ENROLL_REVIEW_ENVIRONMENT'] == true %>">
 
-  <nav class="navbar navbar-default navbar-expand-lg container py-1 px-0">
-    <a class="navbar-brand mr-0" href="#">
+  <nav class="navbar navbar-expand-lg container py-1 px-0">
+    <a class="navbar-brand mr-0" href="#" title="home page link">
```
Comment (kristinmerbach) [inline app/views/ui-components/bs4/v1/navs/_header.html.erb:5]: title should be in translation

File: app/views/shared/_shop_for_plans_progress.html.erb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+<% percent_complete = 0 %>
+<% percent_complete = (step == 1) ? 33 : ((step == 2) ? 66 : ((step == 3) ? 100 : 0)) %>
+
+<h3><%= "#{percent_complete}% Complete" %></h3>
+<ul class="list-unstyled progress-wrapper hidden-xs">
+  <li class="<%= add_progress_class(1, step) %>">
+    <span class="circle-progress"></span>
+    <div class="vertical-line-progress"></div>
+    Plan Selection
```
Comment (kristinmerbach) [inline app/views/shared/_shop_for_plans_progress.html.erb:13]: text into translation here and for the other bullets as well

File: app/views/shared/_sep_shop_for_plans_progress.html.erb [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  <li class="<%= add_progress_class(4, step) %>">
+    <span class="circle-progress"></span>
+    <div class="vertical-line-progress"></div>
+    Complete
+  </li>
+</ul>
+
+<% if step == 2 && (@market_kind == 'shop' || @market_kind == 'fehb') %>
+  <% if @hbx_enrollment and @hbx_enrollment.employee_role.present? %>
+    <%= h(link_to 'Waive Coverage', '#', data: {toggle: "modal", target: "#waive_confirm"}, class: 'btn btn-waive btn-default') %>
```
Comment (kristinmerbach) [inline app/views/shared/_sep_shop_for_plans_progress.html.erb:33]: waive coverage to translation

File: app/views/shared/_sep_shop_for_plans_progress.html.erb [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+<% percent_complete = 0 %>
+<% percent_complete = (step == 1) ? 33 : ((step == 2) ? 66 : ((step == 3) ? 85 : ((step == 4) ? 100: 0 ))) %>
+
+<h3><%= "#{percent_complete}% Complete" %></h3>
+<ul class="list-unstyled progress-wrapper hidden-xs">
+  <li class="<%= add_progress_class(1, step) %>">
+    <span class="circle-progress"></span>
+    <div class="vertical-line-progress"></div>
+    Special Enrollment Period
```
Comment (kristinmerbach) [inline app/views/shared/_sep_shop_for_plans_progress.html.erb:12]: bullets to translations

File: app/views/shared/_consumer_home_address_fields.html.erb [hunk 12]
```diff
@@ -102,13 +102,13 @@
         <% end %>
       </div>
     <% end %>
+
+    <% if f.object.addresses[1].present? %>
+      <span tabindex="0" onkeydown="handleButtonKeyDown(event, 'remove_mailing_address')" class="mb-4 form-action btn-secondary <%= pundit_class Family, :updateable?%>" id="remove_mailing_address" data-cuke="remove_mailing_address">Remove Mailing Address</span>
+    <% else %>
+      <span tabindex="0" onkeydown="handleButtonKeyDown(event, 'add_mailing_address')" id="add_mailing_address" class="mb-4 form-action btn-secondary <%= pundit_class Family, :updateable?%>">Add Mailing Address</span>
```
Comment (kristinmerbach) [inline app/views/shared/_consumer_home_address_fields.html.erb:109]: this has been reworked as pr currently in review, verify these changes against that

**Review by kristinmerbach — CHANGES_REQUESTED** [review]
Tried this locally and the progress menu for the consumer signup ended up messed up, can you please look into?

Also, looks like Add New Member button loads in the form using not @bs4 so looks very wonky on the page

File: db/seedfiles/translations/en/dc/shared.rb [hunk 13]
```diff
@@ -56,8 +55,9 @@
...
...coverage: <br> <ul> <li>Yourself</li><li>Your spouse if you’re legally married (even if you don’t live together)</li><li>Everyone claimed as a dependent on the tax return</li><li>Any child under 21 who lives with you, even if you don’t claim them as a dependent</li><li>Optionally, list any of your children up to age 26, even if they don’t live with you, if you’d like to include them in your health coverage application</li></ul> If you file a federal tax return, include the following people. Include everyone on this list even if some people don’t need health coverage: <br> <ul> <li>Yourself</li> <li> Your spouse if you’re legally married (even if you don’t live together) </li> <li> Anyone under the age of 21 who lives with you</li> <li> Optionally, list any of your children up to age 26, even if they don’t live with you, if you’d like to include them in your health coverage application </li> </ul> You don’t need to list: <ul> <li> Roommates</li> <li> Your unmarried domestic partner (unless you claim this person as your tax dependent)</li> <li> Your spouse if you’re legally separated, divorced, or are a victim of domestic abuse, domestic violence, or spousal abandonment</li> </ul> ",
```
[the remark this answers is not in the document]
Comment (scaustin34) [reply db/seedfiles/translations/en/dc/shared.rb:60]: leaving this for now as its hooked into the 'not sure' modals that use a shared partial that dynamically pulls the text based on the translation key. Will need to figure out a solution for this in the future.

File: app/views/insured/family_members/_family_members.html.erb [hunk 14]
```diff
@@ -1,85 +1,159 @@
...
-    <div class="col-md-3 col-sm-6 no-pd form-group-lg no-pd class-fa-household">
-      <label class="static_label label-floatlabel" for="name">GENDER</label>
-      <span class="field_value floatlabel form-control active-floatlabel" id="name"><%= "#{@person.gender.humanize}" %></span>
+<% if @bs4 %>
+  <% if @change_plan.present? && @sep.blank?%>
+    <h3><%= params[:change_plan] %>, on: <%= @change_plan_date %></h3>
+  <% end %>
+
+  <% if @sep.present? %>
+    <h3><%= @sep.title %> on: <%= @sep.qle_on %></h3>
```
Comment (kristinmerbach) [inline app/views/insured/family_members/_family_members.html.erb:7]: really minor, but on should be in a translation

File: app/views/insured/family_members/_family_members.html.erb [hunk 15]
```diff
@@ -1,85 +1,159 @@
...
+  <div id="dependent_ul">
+    <% @family.active_family_members.reject(&:is_primary_applicant).each_with_index do |dependent, index| %>
+      <%= render "insured/family_members/dependent", dependent: dependent, person: @person, member_number: index+2 %>
+    <% end %>
+  </div>
+
+  <div id="dependent_buttons" class="<%= 'hidden' if @family.active_family_members.empty? %> <%= pundit_class Family, :updateable?%> d-flex flex-column flex-md-row mt-md-5">
+      <%= hidden_field_tag 'origin_source', @source %>
+
+      <%= link_to(main_app.new_insured_family_member_path(:family_id => @family.id, employee_role_id: @employee_role.try(:id), consumer_role_id: @consumer_role.try(:id), bs4: @bs4), :remote => true, :id => "household_info_add_member", class:"button secondary d-flex" ) do %>
```
Comment (kristinmerbach) [inline app/views/insured/family_members/_family_members.html.erb:51]: rework to sanitize, i can show you if needed

File: app/views/insured/plan_shoppings/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,635 @@
...
+          <h1 class="heading-text"><%= l10n("insured.plan_shoppings.show.title") %></h1>
+          <% if @coverage_kind == "health" %>
+            <h4 class="switch-header"><%= l10n("insured.plan_shoppings.show.health_title.content") %></h4>
+          <% else %>
+            <h4 class="switch-header"><%= l10n("insured.plan_shoppings.show.dental_title.content") %></h4>
+          <% end %>
+          <h4 class="switch-header">
+            <% if @market_kind == "shop" && EnrollRegistry.feature_enabled?(:display_cost_warning_message) %>
+              <% if @hbx_enrollment.sponsored_benefit.single_plan_type? %>
+                <strong><%= l10n("Please note your final cost may change based on the final enrollment of all employees.") %></strong>
```
Comment (kristinmerbach) [inline app/views/insured/plan_shoppings/show.html.erb:21]: while this is in an l10n, it should be moved to a translation file

File: app/views/insured/plan_shoppings/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,635 @@
...
+                <b><%= l10n("insured.plan_shoppings.final_cost_change") %></b>
+              <% end %>
+            <% end %>
+          </h4>
+          <div class="row no-buffer">
+            <%= render "shared/plan_shoppings/more_plan_details" %>
+          </div>
+          <p class="pull-right">
+            <% if @market_kind == "individual" || @market_kind == 'coverall' %>
+              <a class="btn btn-default ivl-compare-selected-plans-link" href="javascript:void(0);" style="display: none;" data-cuke="ivl-compare-selected-plans-link"
```
Comment (kristinmerbach) [inline app/views/insured/plan_shoppings/show.html.erb:33]: inline style

File: app/views/insured/plan_shoppings/show.html.erb [hunk 18]
```diff
@@ -0,0 +1,635 @@
...
+              data-uri="<%= comparison_products_plans_path(market_kind: @market_kind, coverage_kind: @coverage_kind) %>">
+                <%= l10n("compare_plans").to_s.upcase %>
+              </a>
+            <% end %>
+            <% if @market_kind == "shop" || @market_kind == 'fehb' %>
+              <a class="btn btn-default compare-selected-plans-link" href="javascript:void(0);" style="display: none;"
```
Comment (kristinmerbach) [inline app/views/insured/plan_shoppings/show.html.erb:39]: same as above

File: app/views/insured/plan_shoppings/show.html.erb [hunk 19]
```diff
@@ -0,0 +1,635 @@
...
+              data-uri="<%= comparison_products_plans_path(market_kind: @market_kind, coverage_kind: @coverage_kind) %>">
+                <%= l10n("compare_plans").to_s.upcase %>
+              </a>
+            <% end %>
+          </p>
+          <div id="plan-summary"></div>
+          <div id="all-plans" class="all-filters-row">
+            <%= render partial: './ui-components/v1/filters/plan_filters' %>
+            <div class="col-lg-9 col-md-9 co-sm-9 col-xs-12" id="planContainer" style="padding-right:0">
```
Comment (kristinmerbach) [inline app/views/insured/plan_shoppings/show.html.erb:48]: inline style can just use pr-0

**Review by kristinmerbach — APPROVED** [review]
Very minor changes, though it would be good if you could look into the XSS warnings throughout

**Pull request merged**